lib/ex_pix_brcode/payments/models/static_pix_payment.ex

```diff
@@ -28,5 +28,17 @@ defmodule ExPixBRCode.Payments.Models.StaticPixPayment do
     |> validate_length(:transaction_amount, max: 13)
     |> validate_format(:transaction_amount, ~r/(^0$)|(^[0-9]+\.[0-9]*$)/)
     |> validate_format(:transaction_id, ~r/^[a-zA-Z0-9]{1,25}$/)
+    |> validate_random_key_format()
+  end
+
+  defp validate_random_key_format(changeset) do
+    key_type = get_change(changeset, :key_type)
+    key = get_change(changeset, :key)
+
+    cond do
+      key_type == "random_key" and key == String.downcase(key) -> changeset
+      key_type == "random_key" and key != String.downcase(key) -> add_error(changeset, :key, "has invalid format")
```

Comment:
```suggestion
      key_type == "random_key" and key == String.downcase(key) -> changeset
      key_type == "random_key" -> add_error(changeset, :key, "has invalid format")
```

To avoid running `String.downcase` twice

Reply:
by the way, I think this is missing the UUID validation, or was it in another function?

Reply:
@polvalente the uuid validation was on a `defp` inside `ExPixBRCode.Payments` module.